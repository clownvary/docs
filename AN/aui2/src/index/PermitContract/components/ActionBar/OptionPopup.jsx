import React from 'react';
import classNames from 'classnames';
import { isIE, isEdge } from 'react-base-ui/lib/utils/browser';
import UIComponent from 'shared/components/UIComponent';
import Alert from 'shared/components/Alert';
import Checkbox from 'react-base-ui/lib/components/Checkbox';
import Tabbable from 'react-base-ui/lib/services/wcag/Tabbable';

import { optionsEnum, optionLabelsEnum } from '../../consts/optionsEnum';

import './options.less';

export default class OptionPopup extends UIComponent {
  defaultOptionState = {
    option: optionsEnum.permit,
    containsRecurring: false,
    showBreakdownFee: false
  };

  constructor(props) {
    super(props);
    this.state = this.defaultOptionState;

    this.bind('handlePopupConfirm');
  }

  open = () => {
    this._refs.alert.open();
  }

  handleOptionSelect = (option) => {
    this.setState({ option });
  };

  handleRecurringOptionChange = () => {
    this.setState(prevState => ({ containsRecurring: !prevState.containsRecurring }));
  }

  handlePopupClose = () => {
    this.setState(this.defaultOptionState);
  };

  handlePopupConfirm() {
    this._refs.alert.onClose();
  }

  handleBreakdownFeeChange = () => {
    this.setState({ showBreakdownFee: !this.state.showBreakdownFee });
  }

  renderOption = (value, index) => {
    const { hasAmendment } = this.props;
    const current = this.state.option === value;
    const label = optionLabelsEnum[value];

    const isDisabled = !hasAmendment && optionsEnum.amendment === value;
    const cls = classNames('options-option', {
      'options-option-selected': current,
      'is-disabled': isDisabled
    });

    return (
      <Tabbable
        aria-label={`option${index} ${label} ${current ? 'checked' : 'unchecked'}`}
        onClick={() => { isDisabled || this.handleOptionSelect(value); }}
      >
        <div className={cls}>
          <div>{label}</div>
          <div>{current && <i className="icon icon-check-circle" />}</div>
        </div>
      </Tabbable>

    );
  };

  renderRecurringOption = () => (
    <div className="options-recurring">
      <Checkbox
        className="options-recurring-checkbox"
        checked={this.state.containsRecurring}
        onChange={this.handleRecurringOptionChange}
        title="heckbox Display"
        aria-label={`checkbox Display all bookings for recurring ${this.state.containsRecurring ? 'selected' : 'unselected'}`}
      >
      Display all bookings for recurring
      </Checkbox>
    </div>
  );

  renderBreakdownFee = () => (
    <div className="options-recurring">
      <Checkbox
        className="options-recurring-checkbox"
        checked={this.state.showBreakdownFee}
        onChange={this.handleBreakdownFeeChange}
        aria-label={`checkbox Show detailed breakdown of fees ${this.state.showBreakdownFee ? 'selected' : 'unselected'}`}
      >
      Show detailed breakdown of fees
      </Checkbox>
    </div>
  );

  renderOptions = () => {
    const { option } = this.state;
    const showRecurringOption = (option === optionsEnum.permit ||
                                 option === optionsEnum.schedule);

    const showBreakdownFeeOption = (option === optionsEnum.permit);

    return (
      <div className={classNames('options', { 'ms-tabbable': isIE() || isEdge() })}>
        {this.renderOption(optionsEnum.permit, 1)}
        {this.renderOption(optionsEnum.schedule, 2)}
        {this.renderOption(optionsEnum.amendment, 3)}
        {showRecurringOption && this.renderRecurringOption()}
        {showBreakdownFeeOption && this.renderBreakdownFee()}
      </div>
    );
  }

  render() {
    const { title, cancelText, confirmText } = this.props;
    return (
      <Alert
        ref={(alert) => { this._refs.alert = alert; }}
        title={title}
        cancelText={cancelText}
        confirmText={confirmText}
        onClose={this.handlePopupClose}
        onCancel={this.handlePopupClose}
        onConfirm={this.handlePopupConfirm}
      >
        {this.renderOptions()}
      </Alert>
    );
  }
}
