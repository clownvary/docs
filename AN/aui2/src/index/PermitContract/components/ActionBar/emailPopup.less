@import '~shared/style/variables/index.less';
@title-color: #313741;
@color-error: #d93a20;
@bg-color-orange: #ff6046;
@bg-color-error: #ffefed;

.emailPopup {
  .btn-strong {
    margin-right: 0;
  }

  &__grid {
    display: flex;
    margin-bottom: 15px;

    :last-child {
      margin-bottom: 0;
    }
  }

  &__label {
    width: 123px;
    margin-right: 20px;
    font-size: 16px;
    font-weight: 600;
    text-align: right;
  }

  &__item {
    &--prompt {
      color: @cyan-blue;
    }

    &--error {
      input,
      textarea {
        color: @color-error;
        background-color: @bg-color-error;
        border: 1px solid @bg-color-orange;
      }
    }

    textarea,
    .input-group {
      width: 447px;
    }

    textarea,
    input {
      font-size: 16px;
    }

    textarea {
      height: 90px;
    }

    input {
      padding-left: 8px;
    }
  }
}
