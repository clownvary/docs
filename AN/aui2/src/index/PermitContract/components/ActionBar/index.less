.actions {
  margin-bottom: 20px;

  button {
    padding: 8px 14px;
    margin-right: 10px;
    font-size: 14px;
    border-radius: 3px;
  }

  a.download-ref {
    display: none;
  }
}
