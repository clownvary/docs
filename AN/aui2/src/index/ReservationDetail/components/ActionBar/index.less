@import '~shared/style/variables/index.less';
/* stylelint-disable */

.actions {
  z-index: 1;
  display: flex;
  justify-content: space-between;
  margin-bottom: 20px;
  background: #fff;

  & > div {
    &:first-child {
      display: flex;
      height: 30px;

      .expiration-date {
        min-width: 230px;

        &-field-label {
          display: inline-block;
          margin-right: 8px;
          font-size: 16px;
          line-height: 28px;
        }

        &-value {
          display: inline-block;
          margin-right: 8px;
          line-height: 28px;

          &.is-expiration {
            color: @state-danger-text;
          }
        }

        &-field {
          display: inline-block;
          font-size: 16px;
          cursor: pointer;
          color: @color-blue;
        }
      }
    }

    &:last-child {
      text-align: right;
      height: 30px;

      .action-item {
        margin: 0 0 10px 8px;
      }
    }
  }

  .btn {
    min-height: 30px;
  }

  button {
    height: 30px;
    min-width: 0;
    min-height: 30px;
    padding: 0 14px;
    font-size: 14px;
    font-weight: normal;
    cursor: pointer;
  }

  .action-item {
    align-items: center;
    width: auto;
    height: 30px;
    margin-right: 10px;
    border-radius: 3px;

    &:last-child {
      margin-right: 0;
      text-decoration: none;
    }

    .aaui-dropdown .aaui-dropdown__menu li {
      font-size: 14px;
    }

    .aaui-button:disabled,
    .aaui-dropdown__button.is-disabled {
      font-size: 14px;

      &,
      .aaui-dropdown__button-text {
        cursor: default;
        color: @loblolly;
        background: @mystic;
        opacity: 1;
      }

      &,
      .icon-chevron-down:before {
        color: @loblolly;
      }

      &:hover {
        background-color: #e6eef1;
        border-color: #ddd;
      }
    }
  }

  .action-status {
    width: 165px;
    padding: 0;
    border: none;

    li {
      font-size: 14px;
      font-weight: normal;
    }
  }

  .action-status {
    .aaui-dropdown__menu-wrapper {
      top: 30px;
    }
  }
}
