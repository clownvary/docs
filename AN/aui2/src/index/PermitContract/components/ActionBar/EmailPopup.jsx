import React from 'react';
import UIComponent from 'shared/components/UIComponent';
import ConfirmAlert from 'shared/components/Alert';
import Alert from 'react-base-ui/lib/components/Alert';
import Input from 'react-base-ui/lib/components/Input';
import TextArea from 'react-base-ui/lib/components/TextArea';

import './emailPopup.less';

export default class EmailPopup extends UIComponent {

  defaultOptionState = { to: '', subject: '', content: '', recipientError: '', showEmailInputPrompt: false };

  emailReg = val => /^[0-9a-z][0-9-._a-z]*@[0-9-.a-z]+\.[a-z]+(?!\W)$/i.test(val);

  constructor(props) {
    super(props);

    this.state = this.defaultOptionState;
  }

  open = () => {
    const { permitNumber, permitLabel = 'Permit', signatures: { customer = {} } } = this.props;
    this.setState(Object.assign({}, this.defaultOptionState, {
      to: customer.email_address || '',
      subject: `${permitLabel} #${permitNumber}`,
      content: `Hello,\nThis is an automated email notification regarding the ${permitLabel} #${permitNumber}.`
    }));

    this._refs.alert.open();
  }

  validateRecipients = recipientString => recipientString.split(',').filter((recipient) => {
    const trimmedRecipient = recipient.trim();
    return trimmedRecipient !== '' && !this.emailReg(trimmedRecipient);
  }).length > 0

  handlePopupConfirm = () => {
    const invalidEmail = this.validateRecipients(this.state.to);

    if (invalidEmail) {
      this.setState({ recipientError: 'The address for Recipients is incorrectly formed.' });
    } else {
      this.props.emailContract({ ...this.state });
      this._refs.alert.onClose();
    }
  }

  handleEmailFocus = ({ target: { value } }) => {
    const { signatures: { customer } } = this.props;

    if (value && value === customer.email_address) {
      this.setState({ to: `${value},` });
    }

    this.setState({ showEmailInputPrompt: true });
  }

  render() {
    const { title, cancelText, confirmText } = this.props;
    const { to, subject, content, recipientError, showEmailInputPrompt } = this.state;

    return (
      <ConfirmAlert
        ref={(alert) => { this._refs.alert = alert; }}
        title={title}
        cancelText={cancelText}
        confirmText={confirmText}
        onConfirm={this.handlePopupConfirm}
        className="emailPopup"
        disableConfirm={!(to.trim() && subject.trim())}
      >
        {
          recipientError && (
            <Alert
              type="error"
              noClose
              className="error-box"
            >
              {recipientError}
            </Alert>
          )
        }
        <div className="emailPopup__grid">
          <div className="emailPopup__label">Recipients</div>
          <div className={`emailPopup__item ${recipientError ? 'emailPopup__item--error' : ''}`}>
            <Input
              type="text"
              value={to}
              aria-label="Recipients"
              onFocus={this.handleEmailFocus}
              onChange={({ target: { value } }) => { this.setState({ to: value, recipientError: '' }); }}
            />
            {
              showEmailInputPrompt && (<div className="emailPopup__item--prompt">* Multiple email addresses separated by commas.</div>)
            }

          </div>
        </div>
        <div className="emailPopup__grid">
          <div className="emailPopup__label emailPopup__label">Email Subject</div>
          <div className="emailPopup__item">
            <Input
              type="text"
              value={subject}
              aria-label="Email Subject"
              maxLength={50}
              onChange={({ target: { value } }) => { this.setState({ subject: value }); }}
            />
          </div>
        </div>
        <div className="emailPopup__grid">
          <div className="emailPopup__label">Email Text</div>
          <div className="emailPopup__item">
            <TextArea
              onChange={({ target: { value } }) => { this.setState({ content: value }); }}
              value={content}
              aria-label="Email Text"
            />
          </div>
        </div>
      </ConfirmAlert>
    );
  }
}
