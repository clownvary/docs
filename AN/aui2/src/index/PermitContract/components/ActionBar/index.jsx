import React from 'react';
import UIComponent from 'shared/components/UIComponent';
import Button from 'react-base-ui/lib/components/Button';
import PrintOptionPopup from './PrintOptionPopup';
import PdfOptionPopup from './PdfOptionPopup';
import EmailPopup from './EmailPopup';
import './index.less';

export default class ActionBar extends UIComponent {

  onSaveAsPdfBtnClick = () => {
    const signatureSection = document.querySelector('.signature');
    if (signatureSection) {
      signatureSection.style.height = `${signatureSection.offsetHeight}px`;
    }
    this._refs.pdfOptionPopup.open();
  }

  render() {
    const { hasAmendment, permitId, permitNumber, emailContract, savePdfAction,
      permitLabel, showEmail, signatures } = this.props;

    return (
      <div className="actions">
        <Button
          className="btn action-item"
          size="sm"
          onClick={() => { this._refs.printOptionPopup.open(); }}
        >
          Print
        </Button>

        {
          showEmail && (
            <Button
              className="btn action-item"
              size="sm"
              onClick={() => { this._refs.emailPopup.open(); }}
            >
              Send Email
            </Button>
          )
        }

        <Button
          className="btn action-item"
          size="sm"
          onClick={this.onSaveAsPdfBtnClick}
        >
          Save as PDF
        </Button>

        <PrintOptionPopup
          title="Print Options"
          confirmText="Print"
          cancelText="Cancel"
          hasAmendment={hasAmendment}
          ref={(c) => { this._refs.printOptionPopup = c; }}
        />

        <EmailPopup
          title="Email Detail"
          confirmText="Send"
          cancelText="Cancel"
          ref={(c) => { this._refs.emailPopup = c; }}
          permitNumber={permitNumber}
          permitId={permitId}
          permitLabel={permitLabel}
          emailContract={emailContract}
          signatures={signatures}
        />

        <PdfOptionPopup
          ref={(c) => { this._refs.pdfOptionPopup = c; }}
          title="Save PDF"
          confirmText="Save"
          cancelText="Cancel"
          hasAmendment={hasAmendment}
          savePdfAction={savePdfAction}
          permitNumber={permitNumber}
        />
      </div>
    );
  }
}
