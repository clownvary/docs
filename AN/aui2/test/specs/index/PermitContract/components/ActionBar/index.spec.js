import React from 'react';
import { mount } from 'enzyme';
import toJson from 'enzyme-to-json';
import ActionBar from 'index/PermitContract/components/ActionBar/index';

describe('index/PermitContract/components/ActionBar/index', () => {
  test('should render without errors', () => {
    const component = mount(<ActionBar showEmail signatures={{}}  />);
    expect(toJson(component)).toMatchSnapshot();
  });

  test('print and save-pdf works correctly', () => {
    const component = mount(<ActionBar showEmail signatures={{}}  />);

    const printBtn = component.find('button').at(0);
    const emailBtn = component.find('button').at(1);
    const savePdfBtn = component.find('button').at(2);

    const printModal = component.find('.modal').at(0);
    const emailModal = component.find('.modal').at(1);
    const savePdfModal = component.find('.modal').at(2);

    expect(component.find('.modal-wrap').length).toBe(3);
    expect(printModal.hasClass('is-open')).toBe(false);
    expect(emailModal.hasClass('is-open')).toBe(false);
    expect(savePdfBtn.hasClass('is-open')).toBe(false);

    printBtn.simulate('click');
    expect(printModal.hasClass('is-open')).toEqual(true);

    emailBtn.simulate('click');
    expect(emailModal.hasClass('is-open')).toEqual(true);

    savePdfBtn.simulate('click');
    expect(savePdfModal.hasClass('is-open')).toEqual(true);
  });
});
