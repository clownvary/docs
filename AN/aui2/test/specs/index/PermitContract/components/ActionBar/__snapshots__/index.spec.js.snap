// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`index/PermitContract/components/ActionBar/index should render without errors 1`] = `
<ActionBar
  showEmail={true}
  signatures={Object {}}
>
  <div
    className="actions"
  >
    <Button
      className="btn action-item"
      loading={false}
      menuData={Array []}
      noSubmit={false}
      onClick={[Function]}
      size="sm"
      type="secondary"
    >
      <button
        className="btn btn-secondary btn--sm btn action-item"
        onClick={[Function]}
        onMouseEnter={[Function]}
        onMouseLeave={[Function]}
        onMouseOver={[Function]}
        type="submit"
      >
        <span>
          Print
        </span>
      </button>
    </Button>
    <Button
      className="btn action-item"
      loading={false}
      menuData={Array []}
      noSubmit={false}
      onClick={[Function]}
      size="sm"
      type="secondary"
    >
      <button
        className="btn btn-secondary btn--sm btn action-item"
        onClick={[Function]}
        onMouseEnter={[Function]}
        onMouseLeave={[Function]}
        onMouseOver={[Function]}
        type="submit"
      >
        <span>
          Send Email
        </span>
      </button>
    </Button>
    <Button
      className="btn action-item"
      loading={false}
      menuData={Array []}
      noSubmit={false}
      onClick={[Function]}
      size="sm"
      type="secondary"
    >
      <button
        className="btn btn-secondary btn--sm btn action-item"
        onClick={[Function]}
        onMouseEnter={[Function]}
        onMouseLeave={[Function]}
        onMouseOver={[Function]}
        type="submit"
      >
        <span>
          Save as PDF
        </span>
      </button>
    </Button>
    <PrintOptionPopup
      cancelText="Cancel"
      confirmText="Print"
      title="Print Options"
    >
      <Alert
        cancelText="Cancel"
        confirmText="Print"
        onCancel={[Function]}
        onClose={[Function]}
        onConfirm={[Function]}
        title="Print Options"
      >
        <Modal
          className="modal--alert undefined"
          onClose={[Function]}
          shown={false}
          title="Print Options"
        >
          <div
            className="modal-wrap"
            role="dialog"
          >
            <div
              className="modal modal--alert undefined"
              onKeyDown={[Function]}
            >
              <section
                className="modal-box"
              >
                <header
                  className="modal-header"
                >
                  <h3
                    className="modal-title"
                  >
                    Print Options
                  </h3>
                  <span
                    aria-label="close-dialog"
                    className="icon icon-close modal-close"
                    onClick={[Function]}
                    onKeyDown={[Function]}
                    role="button"
                    tabIndex="0"
                  />
                </header>
                <div
                  className="modal-body"
                >
                  <div
                    className="options"
                  >
                    <Tabbable
                      aria-label="option1 undefined checked"
                      onClick={[Function]}
                      prefixCls="an-tabbable"
                    >
                      <span
                        className="an-tabbable"
                        onClick={[Function]}
                      >
                        <a
                          aria-label="option1 undefined checked"
                          className="an-tabbable__fake-link"
                          href="javascript:void(0)"
                          onClick={[Function]}
                          role="button"
                        >
                          <div
                            className="options-option options-option-selected"
                          >
                            <div />
                            <div>
                              <i
                                className="icon icon-check-circle"
                              />
                            </div>
                          </div>
                          <span
                            style={
                              Object {
                                "display": "none",
                              }
                            }
                          >
                            wcag_hidden_text
                          </span>
                        </a>
                      </span>
                    </Tabbable>
                    <Tabbable
                      aria-label="option2 Schedule unchecked"
                      onClick={[Function]}
                      prefixCls="an-tabbable"
                    >
                      <span
                        className="an-tabbable"
                        onClick={[Function]}
                      >
                        <a
                          aria-label="option2 Schedule unchecked"
                          className="an-tabbable__fake-link"
                          href="javascript:void(0)"
                          onClick={[Function]}
                          role="button"
                        >
                          <div
                            className="options-option"
                          >
                            <div>
                              Schedule
                            </div>
                            <div />
                          </div>
                          <span
                            style={
                              Object {
                                "display": "none",
                              }
                            }
                          >
                            wcag_hidden_text
                          </span>
                        </a>
                      </span>
                    </Tabbable>
                    <Tabbable
                      aria-label="option3 Amendment History unchecked"
                      onClick={[Function]}
                      prefixCls="an-tabbable"
                    >
                      <span
                        className="an-tabbable"
                        onClick={[Function]}
                      >
                        <a
                          aria-label="option3 Amendment History unchecked"
                          className="an-tabbable__fake-link"
                          href="javascript:void(0)"
                          onClick={[Function]}
                          role="button"
                        >
                          <div
                            className="options-option is-disabled"
                          >
                            <div>
                              Amendment History
                            </div>
                            <div />
                          </div>
                          <span
                            style={
                              Object {
                                "display": "none",
                              }
                            }
                          >
                            wcag_hidden_text
                          </span>
                        </a>
                      </span>
                    </Tabbable>
                    <div
                      className="options-recurring"
                    >
                      <Checkbox
                        aria-label="checkbox Display all bookings for recurring unselected"
                        checked={false}
                        className="options-recurring-checkbox"
                        onChange={[Function]}
                        title="heckbox Display"
                      >
                        <CheckboxComponent
                          aria-label="checkbox Display all bookings for recurring unselected"
                          checked={false}
                          className="options-recurring-checkbox"
                          onChange={[Function]}
                          title="heckbox Display"
                        >
                          <label
                            className="checkbox-wrapper options-recurring-checkbox"
                            htmlFor="checkbox_attr_for_1"
                          >
                            <span
                              className="checkbox"
                            >
                              <input
                                aria-checked={false}
                                aria-label="checkbox Display all bookings for recurring unselected"
                                checked={false}
                                className="checkbox__input"
                                id="checkbox_attr_for_1"
                                onChange={[Function]}
                                role="checkbox"
                                title="heckbox Display"
                                type="checkbox"
                              />
                              <span
                                className="checkbox__inner"
                              />
                              <span
                                className="checkbox__text"
                              >
                                Display all bookings for recurring
                              </span>
                            </span>
                          </label>
                        </CheckboxComponent>
                      </Checkbox>
                    </div>
                    <div
                      className="options-recurring"
                    >
                      <Checkbox
                        aria-label="checkbox Show detailed breakdown of fees unselected"
                        checked={false}
                        className="options-recurring-checkbox"
                        onChange={[Function]}
                      >
                        <CheckboxComponent
                          aria-label="checkbox Show detailed breakdown of fees unselected"
                          checked={false}
                          className="options-recurring-checkbox"
                          onChange={[Function]}
                        >
                          <label
                            className="checkbox-wrapper options-recurring-checkbox"
                            htmlFor="checkbox_attr_for_2"
                          >
                            <span
                              className="checkbox"
                            >
                              <input
                                aria-checked={false}
                                aria-label="checkbox Show detailed breakdown of fees unselected"
                                checked={false}
                                className="checkbox__input"
                                id="checkbox_attr_for_2"
                                onChange={[Function]}
                                role="checkbox"
                                type="checkbox"
                              />
                              <span
                                className="checkbox__inner"
                              />
                              <span
                                className="checkbox__text"
                              >
                                Show detailed breakdown of fees
                              </span>
                            </span>
                          </label>
                        </CheckboxComponent>
                      </Checkbox>
                    </div>
                  </div>
                </div>
                <div
                  className="modal-footer"
                >
                  <Button
                    className=""
                    loading={false}
                    menuData={Array []}
                    noSubmit={false}
                    onClick={[Function]}
                    type="secondary"
                  >
                    <button
                      className="btn btn-secondary"
                      onClick={[Function]}
                      onMouseEnter={[Function]}
                      onMouseLeave={[Function]}
                      onMouseOver={[Function]}
                      type="submit"
                    >
                      <span>
                        Cancel
                      </span>
                    </button>
                  </Button>
                  <Button
                    className=""
                    loading={false}
                    menuData={Array []}
                    noSubmit={false}
                    onClick={[Function]}
                    type="strong"
                  >
                    <button
                      className="btn btn-strong"
                      onClick={[Function]}
                      onMouseEnter={[Function]}
                      onMouseLeave={[Function]}
                      onMouseOver={[Function]}
                      type="submit"
                    >
                      <span>
                        Print
                      </span>
                    </button>
                  </Button>
                </div>
              </section>
            </div>
            <div
              className="modal-mask"
            />
          </div>
        </Modal>
      </Alert>
    </PrintOptionPopup>
    <EmailPopup
      cancelText="Cancel"
      confirmText="Send"
      signatures={Object {}}
      title="Email Detail"
    >
      <Alert
        cancelText="Cancel"
        className="emailPopup"
        confirmText="Send"
        disableConfirm={true}
        onConfirm={[Function]}
        title="Email Detail"
      >
        <Modal
          className="modal--alert emailPopup"
          onClose={[Function]}
          shown={false}
          title="Email Detail"
        >
          <div
            className="modal-wrap"
            role="dialog"
          >
            <div
              className="modal modal--alert emailPopup"
              onKeyDown={[Function]}
            >
              <section
                className="modal-box"
              >
                <header
                  className="modal-header"
                >
                  <h3
                    className="modal-title"
                  >
                    Email Detail
                  </h3>
                  <span
                    aria-label="close-dialog"
                    className="icon icon-close modal-close"
                    onClick={[Function]}
                    onKeyDown={[Function]}
                    role="button"
                    tabIndex="0"
                  />
                </header>
                <div
                  className="modal-body"
                >
                  <div
                    className="emailPopup__grid"
                  >
                    <div
                      className="emailPopup__label"
                    >
                      Recipients
                    </div>
                    <div
                      className="emailPopup__item "
                    >
                      <Input
                        aria-label="Recipients"
                        icon={false}
                        inputRef={[Function]}
                        onChange={[Function]}
                        onFocus={[Function]}
                        size="md"
                        type="text"
                        value=""
                      >
                        <div
                          className="input-group input-group--md"
                        >
                          <input
                            aria-label="Recipients"
                            className="input input-group__field input__field"
                            onBlur={[Function]}
                            onChange={[Function]}
                            onFocus={[Function]}
                            onInput={[Function]}
                            onKeyPress={[Function]}
                            onPaste={[Function]}
                            type="text"
                            value=""
                          />
                        </div>
                      </Input>
                    </div>
                  </div>
                  <div
                    className="emailPopup__grid"
                  >
                    <div
                      className="emailPopup__label emailPopup__label"
                    >
                      Email Subject
                    </div>
                    <div
                      className="emailPopup__item"
                    >
                      <Input
                        aria-label="Email Subject"
                        icon={false}
                        inputRef={[Function]}
                        maxLength={50}
                        onChange={[Function]}
                        size="md"
                        type="text"
                        value=""
                      >
                        <div
                          className="input-group input-group--md"
                        >
                          <input
                            aria-label="Email Subject"
                            className="input input-group__field input__field"
                            maxLength={50}
                            onBlur={[Function]}
                            onChange={[Function]}
                            onInput={[Function]}
                            onKeyPress={[Function]}
                            onPaste={[Function]}
                            type="text"
                            value=""
                          />
                        </div>
                      </Input>
                    </div>
                  </div>
                  <div
                    className="emailPopup__grid"
                  >
                    <div
                      className="emailPopup__label"
                    >
                      Email Text
                    </div>
                    <div
                      className="emailPopup__item"
                    >
                      <TextArea
                        aria-label="Email Text"
                        className="input"
                        onChange={[Function]}
                        value=""
                      >
                        <textarea
                          aria-label="Email Text"
                          className="input"
                          onChange={[Function]}
                          value=""
                        />
                      </TextArea>
                    </div>
                  </div>
                </div>
                <div
                  className="modal-footer"
                >
                  <Button
                    className=""
                    loading={false}
                    menuData={Array []}
                    noSubmit={false}
                    onClick={[Function]}
                    type="secondary"
                  >
                    <button
                      className="btn btn-secondary"
                      onClick={[Function]}
                      onMouseEnter={[Function]}
                      onMouseLeave={[Function]}
                      onMouseOver={[Function]}
                      type="submit"
                    >
                      <span>
                        Cancel
                      </span>
                    </button>
                  </Button>
                  <Button
                    className=""
                    disabled={true}
                    loading={false}
                    menuData={Array []}
                    noSubmit={false}
                    onClick={[Function]}
                    type="strong"
                  >
                    <button
                      className="btn btn-strong"
                      disabled={true}
                      onClick={[Function]}
                      onMouseEnter={[Function]}
                      onMouseLeave={[Function]}
                      onMouseOver={[Function]}
                      type="submit"
                    >
                      <span>
                        Send
                      </span>
                    </button>
                  </Button>
                </div>
              </section>
            </div>
            <div
              className="modal-mask"
            />
          </div>
        </Modal>
      </Alert>
    </EmailPopup>
    <PdfOptionPopup
      cancelText="Cancel"
      confirmText="Save"
      title="Save PDF"
    >
      <Alert
        cancelText="Cancel"
        confirmText="Save"
        onCancel={[Function]}
        onClose={[Function]}
        onConfirm={[Function]}
        title="Save PDF"
      >
        <Modal
          className="modal--alert undefined"
          onClose={[Function]}
          shown={false}
          title="Save PDF"
        >
          <div
            className="modal-wrap"
            role="dialog"
          >
            <div
              className="modal modal--alert undefined"
              onKeyDown={[Function]}
            >
              <section
                className="modal-box"
              >
                <header
                  className="modal-header"
                >
                  <h3
                    className="modal-title"
                  >
                    Save PDF
                  </h3>
                  <span
                    aria-label="close-dialog"
                    className="icon icon-close modal-close"
                    onClick={[Function]}
                    onKeyDown={[Function]}
                    role="button"
                    tabIndex="0"
                  />
                </header>
                <div
                  className="modal-body"
                >
                  <div
                    className="options"
                  >
                    <Tabbable
                      aria-label="option1 undefined checked"
                      onClick={[Function]}
                      prefixCls="an-tabbable"
                    >
                      <span
                        className="an-tabbable"
                        onClick={[Function]}
                      >
                        <a
                          aria-label="option1 undefined checked"
                          className="an-tabbable__fake-link"
                          href="javascript:void(0)"
                          onClick={[Function]}
                          role="button"
                        >
                          <div
                            className="options-option options-option-selected"
                          >
                            <div />
                            <div>
                              <i
                                className="icon icon-check-circle"
                              />
                            </div>
                          </div>
                          <span
                            style={
                              Object {
                                "display": "none",
                              }
                            }
                          >
                            wcag_hidden_text
                          </span>
                        </a>
                      </span>
                    </Tabbable>
                    <Tabbable
                      aria-label="option2 Schedule unchecked"
                      onClick={[Function]}
                      prefixCls="an-tabbable"
                    >
                      <span
                        className="an-tabbable"
                        onClick={[Function]}
                      >
                        <a
                          aria-label="option2 Schedule unchecked"
                          className="an-tabbable__fake-link"
                          href="javascript:void(0)"
                          onClick={[Function]}
                          role="button"
                        >
                          <div
                            className="options-option"
                          >
                            <div>
                              Schedule
                            </div>
                            <div />
                          </div>
                          <span
                            style={
                              Object {
                                "display": "none",
                              }
                            }
                          >
                            wcag_hidden_text
                          </span>
                        </a>
                      </span>
                    </Tabbable>
                    <Tabbable
                      aria-label="option3 Amendment History unchecked"
                      onClick={[Function]}
                      prefixCls="an-tabbable"
                    >
                      <span
                        className="an-tabbable"
                        onClick={[Function]}
                      >
                        <a
                          aria-label="option3 Amendment History unchecked"
                          className="an-tabbable__fake-link"
                          href="javascript:void(0)"
                          onClick={[Function]}
                          role="button"
                        >
                          <div
                            className="options-option is-disabled"
                          >
                            <div>
                              Amendment History
                            </div>
                            <div />
                          </div>
                          <span
                            style={
                              Object {
                                "display": "none",
                              }
                            }
                          >
                            wcag_hidden_text
                          </span>
                        </a>
                      </span>
                    </Tabbable>
                    <div
                      className="options-recurring"
                    >
                      <Checkbox
                        aria-label="checkbox Display all bookings for recurring unselected"
                        checked={false}
                        className="options-recurring-checkbox"
                        onChange={[Function]}
                        title="heckbox Display"
                      >
                        <CheckboxComponent
                          aria-label="checkbox Display all bookings for recurring unselected"
                          checked={false}
                          className="options-recurring-checkbox"
                          onChange={[Function]}
                          title="heckbox Display"
                        >
                          <label
                            className="checkbox-wrapper options-recurring-checkbox"
                            htmlFor="checkbox_attr_for_3"
                          >
                            <span
                              className="checkbox"
                            >
                              <input
                                aria-checked={false}
                                aria-label="checkbox Display all bookings for recurring unselected"
                                checked={false}
                                className="checkbox__input"
                                id="checkbox_attr_for_3"
                                onChange={[Function]}
                                role="checkbox"
                                title="heckbox Display"
                                type="checkbox"
                              />
                              <span
                                className="checkbox__inner"
                              />
                              <span
                                className="checkbox__text"
                              >
                                Display all bookings for recurring
                              </span>
                            </span>
                          </label>
                        </CheckboxComponent>
                      </Checkbox>
                    </div>
                    <div
                      className="options-recurring"
                    >
                      <Checkbox
                        aria-label="checkbox Show detailed breakdown of fees unselected"
                        checked={false}
                        className="options-recurring-checkbox"
                        onChange={[Function]}
                      >
                        <CheckboxComponent
                          aria-label="checkbox Show detailed breakdown of fees unselected"
                          checked={false}
                          className="options-recurring-checkbox"
                          onChange={[Function]}
                        >
                          <label
                            className="checkbox-wrapper options-recurring-checkbox"
                            htmlFor="checkbox_attr_for_4"
                          >
                            <span
                              className="checkbox"
                            >
                              <input
                                aria-checked={false}
                                aria-label="checkbox Show detailed breakdown of fees unselected"
                                checked={false}
                                className="checkbox__input"
                                id="checkbox_attr_for_4"
                                onChange={[Function]}
                                role="checkbox"
                                type="checkbox"
                              />
                              <span
                                className="checkbox__inner"
                              />
                              <span
                                className="checkbox__text"
                              >
                                Show detailed breakdown of fees
                              </span>
                            </span>
                          </label>
                        </CheckboxComponent>
                      </Checkbox>
                    </div>
                  </div>
                </div>
                <div
                  className="modal-footer"
                >
                  <Button
                    className=""
                    loading={false}
                    menuData={Array []}
                    noSubmit={false}
                    onClick={[Function]}
                    type="secondary"
                  >
                    <button
                      className="btn btn-secondary"
                      onClick={[Function]}
                      onMouseEnter={[Function]}
                      onMouseLeave={[Function]}
                      onMouseOver={[Function]}
                      type="submit"
                    >
                      <span>
                        Cancel
                      </span>
                    </button>
                  </Button>
                  <Button
                    className=""
                    loading={false}
                    menuData={Array []}
                    noSubmit={false}
                    onClick={[Function]}
                    type="strong"
                  >
                    <button
                      className="btn btn-strong"
                      onClick={[Function]}
                      onMouseEnter={[Function]}
                      onMouseLeave={[Function]}
                      onMouseOver={[Function]}
                      type="submit"
                    >
                      <span>
                        Save
                      </span>
                    </button>
                  </Button>
                </div>
              </section>
            </div>
            <div
              className="modal-mask"
            />
          </div>
        </Modal>
      </Alert>
    </PdfOptionPopup>
  </div>
</ActionBar>
`;
